File: C#/Insercion_inicial.cs
using System;
using System.Collections.Generic;

class Program {
    static void Main() {
        //uso una lista para poder insertar elementos en el array
        List<int> insercion = new List<int> { 11, 21, 31, 41, 51, 61 };
        int ele = 52; 

        int largo = insercion.Count; //variable que guarda el largo del arreglo

        //Recorro la lista para mostrar los elementos
        Console.Write("Antes de la insercion, el array es: ");
        for (int indice = 0; indice < largo; indice++) {
            Console.Write(insercion[indice] + " ");
        }
        Console.WriteLine();

        //inserta el elemento en el índice 0
        insercion.Insert(0, ele);
        
        //recorre el arreglo con la insercion hecha
        Console.Write("Después de la eliminacion, el array es: ");
        for (int indice = 0; indice < largo; indice++) {
            Console.Write(insercion[indice] + " ");
        }
        Console.WriteLine();
    } 
}

File: C++/Insercion_inicial.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    //uso vector para poder insertar elementos a un arreglo
    vector<int> insercion = {11, 21, 31, 41, 51, 61};
    int ele = 52;
    
    cout << "Antes de la insercion, el array es: ";
    for (int indice = 0; indice < insercion.size(); indice++) {
        cout << insercion[indice] << " ";
    }
    cout << endl;
    
    //inserta el elemento en el inicio del array (begin = indice 0)
    insercion.insert(insercion.begin(), ele);
    
    cout << "Después de la eliminacion, el array es: ";
    for (int indice = 0; indice < insercion.size(); indice++) {
        cout << insercion[indice] << " ";
    }
    cout << endl;
    
    return 0;
}